src/Uri.php

```diff
@@ -254,16 +252,20 @@ private function buildPrimaryIdentifier()
     {
         $identifier = '';
 
-        if ($this->parts['scheme']) {
+        if ($this->parts['scheme'] !== '') {
             $identifier .= $this->parts['scheme'].'://';
         }
 
         $identifier .= $this->authority.$this->parts['path'];
 
-        if ($this->parts['query']) {
+        if ($this->parts['query'] !== '') {
             $identifier .= '?'.$this->parts['query'];
         }
 
+        if ($this->parts['fragment'] !== '' && $this->parts['fragment'][0] !== '/') {
```

Comment:
Good point, I was unaware of the RFC 3986 terminology. The purpose of this is that we need non-JSON-pointer fragments to be used as part of schema identifiers.